### ISSUE TITLE
Issue with query validation?

### ISSUE THREAD

**Issue by njlr** [Zaid-Ajaj/Snowflaqe#9]
My schema:

```gql
type Foo {
  id: ID!
  name: String!
}

type Query {
  foo(id: ID!): Foo
}
```

My query:

```gql
query($id: ID!) {
  foo(id: $id) {
    id
    name
  }
}
```

But I get a validation error:

> Field foo contains type mismatch for argument id of type ID! which references variable $id of type ID!

This is unexpected since `id` and `$id` have the same type (`ID!`).

**Comment by njlr** [comment]
It works if I change `ID!` to `String!`, so it's probably just a missing case somewhere.

**Comment by njlr** [comment]
If I had to guess, it's probably here: https://github.com/Zaid-Ajaj/Snowflaqe/blob/14449d5343688bbe16b6eabcba9b4baa08905528/src/Query.fs#L329 (no case for `"ID"`)

**Comment by Zaid-Ajaj** [comment]
Hi @njlr, thanks a lot for filing the issue with the relevant debug information. Indeed I missed a case in the validation code but it should be fixed as of v1.5.1 :smile:
